source/common/upstream/health_discovery_service.cc

```diff
@@ -6,19 +6,31 @@ namespace Envoy {
 namespace Upstream {
 
 HdsDelegate::HdsDelegate(const envoy::api::v2::core::Node& node, Stats::Scope& scope,
-                         Grpc::AsyncClientPtr async_client, Event::Dispatcher& dispatcher)
+                         Grpc::AsyncClientPtr async_client, Event::Dispatcher& dispatcher,
+                         Runtime::Loader& runtime, Envoy::Stats::Store& stats,
+                         Ssl::ContextManager& ssl_context_manager,
+                         Secret::SecretManager& secret_manager, Runtime::RandomGenerator& random,
+                         ClusterInfoFactory& info_factory,
+                         AccessLog::AccessLogManager& access_log_manager)
     : stats_{ALL_HDS_STATS(POOL_COUNTER_PREFIX(scope, "hds_delegate."))},
       async_client_(std::move(async_client)),
       service_method_(*Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
-          "envoy.service.discovery.v2.HealthDiscoveryService.StreamHealthCheck")) {
+          "envoy.service.discovery.v2.HealthDiscoveryService.StreamHealthCheck")),
+      runtime_(runtime), store_stats(stats), ssl_context_manager_(ssl_context_manager),
+      secret_manager_(secret_manager), random_(random), dispatcher_(dispatcher),
+      info_factory_(info_factory), access_log_manager_(access_log_manager) {
   health_check_request_.mutable_node()->MergeFrom(node);
   retry_timer_ = dispatcher.createTimer([this]() -> void { establishNewStream(); });
-  response_timer_ = dispatcher.createTimer([this]() -> void { sendHealthCheckRequest(); });
+  server_response_timer_ = dispatcher.createTimer([this]() -> void { sendResponse(); });
```

Comment:
I would rename these two timers to `hds_retry_timer_` and `hds_stream_response_timer_` respectively to make it clearer what they do.